torch/_utils_internal.py

```diff
@@ -33,5 +34,23 @@ def resolve_library_path(path):
     return os.path.realpath(path)
 
 
+def get_source_lines_and_file(obj):
+    """
+    Wrapper around inspect.getsourcelines and inspect.getsourcefile.
+
+    Returns: (sourcelines, file_lino, filename)
+    """
+    filename = None  # in case getsourcefile throws
+    try:
+        filename = inspect.getsourcefile(obj)
```

Comment:
It's rare but I've run into it before, this can throw a `TypeError` if `obj` is a built-in, https://docs.python.org/3/library/inspect.html#inspect.getsourcefile, we should catch that too

Reply:
I think that's fine - I only want to enhance error message if source is not available. I guess if it's a builtin - it'd be either a problem with user input explicitly (like `torch.jit.script(open)`) or a bug in jit. Default message is pretty ok (`TypeError: module, class, method, function, traceback, frame, or code object was expected, got builtin_function_or_method`)